relay-general/src/protocol/replay.rs

```diff
@@ -0,0 +1,567 @@
+            )]),
+            trace_ids: Annotated::new(vec![Annotated::new(
+                "52df9022835246eeb317dbd739ccd059".to_string(),
+            )]),
+            platform: Annotated::new("myplatform".to_string()),
+            release: Annotated::new("myrelease".to_string().into()),
+            dist: Annotated::new("mydist".to_string()),
+            environment: Annotated::new("myenv".to_string()),
+            tags: {
+                let items = vec![Annotated::new(TagEntry(
+                    Annotated::new("tag".to_string()),
+                    Annotated::new("value".to_string()),
+                ))];
+                Annotated::new(Tags(items.into()))
+            },
+            ..Default::default()
+        });
+
+        assert_eq!(replay, Annotated::from_json(json).unwrap());
+    }
+
+    #[test]
+    fn test_lenient_release() {
+        let input = r#"{"release":42}"#;
+        let output = r#"{"release":"42"}"#;
+        let event = Annotated::new(Replay {
+            release: Annotated::new("42".to_string().into()),
+            ..Default::default()
+        });
+
+        assert_eq!(event, Annotated::from_json(input).unwrap());
+        assert_eq!(output, event.to_json().unwrap());
+    }
+
+    #[test]
+    fn test_set_user_agent_meta() {
+        let os_context = Annotated::new(ContextInner(Context::Os(Box::new(OsContext {
+            name: Annotated::new("Mac OS X".to_string()),
+            version: Annotated::new("10.15.7".to_string()),
+            ..Default::default()
+        }))));
+        let browser_context =
+            Annotated::new(ContextInner(Context::Browser(Box::new(BrowserContext {
+                name: Annotated::new("Safari".to_string()),
+                version: Annotated::new("15.5".to_string()),
+                ..Default::default()
+            }))));
+        let device_context =
+            Annotated::new(ContextInner(Context::Device(Box::new(DeviceContext {
+                family: Annotated::new("Mac".to_string()),
+                brand: Annotated::new("Apple".to_string()),
+                model: Annotated::new("Mac".to_string()),
+                ..Default::default()
+            }))));
+
+        // Parse user input.
+        let payload = include_str!("../../tests/fixtures/replays/replay.json");
+
+        let mut replay: Annotated<Replay> = Annotated::from_json(payload).unwrap();
+        let replay_value = replay.value_mut().as_mut().unwrap();
+        replay_value.normalize(None, None);
+
+        let loaded_browser_context = replay_value
+            .contexts
+            .value_mut()
+            .as_mut()
+            .unwrap()
+            .get("browser")
+            .unwrap()
+            .clone();
+
+        let loaded_os_context = replay_value
+            .contexts
+            .value_mut()
+            .as_mut()
+            .unwrap()
+            .get("client_os")
+            .unwrap()
+            .clone();
```

Comment:
No need to take a mutable reference or clone here, please simplify to 

```suggestion
            .value()
            .unwrap()
            .get("client_os")
            .unwrap();
```

and then compare to a reference in the assert (i.e. `&os_context`).